src/Messages/TSCall.h

```diff
@@ -13,6 +13,9 @@ typedef enum {
     RPRecentCallTypeIncoming = 1,
     RPRecentCallTypeOutgoing,
     RPRecentCallTypeMissed,
+    // These call types are used until the call connects.
+    RPRecentCallTypeOutgoingIncomplete,
+    RPRecentCallTypeIncomingIncomplete,
```

Comment:
Incoming calls start as "incoming incomplete" and end up as "incoming" or "missed."
Outgoing calls start as "outgoing incomplete" and only become "outgoing" when they connect. If the outgoing call never connects (because you hang up first or the callee declines) the call simply remains in "outgoing incomplete".  It would be more consistent to rename:

* RPRecentCallTypeIncoming -> RPRecentCallTypeIncomingComplete
* RPRecentCallTypeOutgoing -> RPRecentCallTypeOutgoingComplete
* RPRecentCallTypeMissed -> RPRecentCallTypeIncomingFailed

And make a new state RPRecentCallTypeOutgoingFailed, but I didn't bother since this wouldn't have any visible effect.